imagetest/test_suites/metadata/setup.go

```diff
@@ -12,40 +12,189 @@ var Name = "metadata"
 
 const (
 	daemonScriptTemplate = `#!/bin/bash
-
-nohup sleep 3600 > /dev/null 2>&1 < /dev/null &
-echo $! > %s
-`
+	nohup sleep 3600 > /dev/null 2>&1 < /dev/null &
+	echo $! > %s
+	`
 	daemonOutputPath = "/daemon_out.txt"
 
-	startupScriptTemplate = `#!/bin/bash
-echo "%s" > %s`
-	startupOutputPath  = "/startup_out.txt"
-	startupContent     = "The startup script worked."
-	shutdownScriptTime = `#!/bin/bash
-
-while [[ 1 ]]; do
-  date +%s >> /shutdown.txt
-  sync
-  sleep 1
-done`
-	shutdownScriptTemplate = `#!/bin/bash
-echo "%s" > %s`
-	shutdownOutputPath = "/shutdown_out.txt"
-	shutdownContent    = "The shutdown script worked."
+	bashScriptTemplate = `#!/bin/bash
+	echo "%s" > %s`
+	windowsScriptTemplate = `echo "%s" > %s
+	ping -n 2 127.0.0.1 >NUL`
+	powershellScriptTemplate = `Set-Content -Value "%s" -Path %s
+	Start-Sleep 1`
+	startupContent  = "The startup script worked."
+	shutdownContent = "The shutdown script worked."
+
+	timeOutputPath        = "/shutdown.txt"
+	windowsTimeOutputPath = "C:\\shutdown.txt"
+
+	bashTimeScriptTemplate = `#!/bin/bash
+	while [[ 1 ]]; do
+	date +%%s >> %s
+	  sync
+	  sleep 1
+	  done`
+
+	windowsTimeScriptTemplate = `while ($true) {
+		$time = (Get-Date).ToString("yyyyMMdd-HHMMss")
+		Add-Content -Path %s -Value $time
+		Start-Sleep  1
+		}`
+
 	// max metadata value 256kb https://cloud.google.com/compute/docs/metadata/setting-custom-metadata#limitations
 	// metadataMaxLength = 256 * 1024
 	// TODO(hopkiw): above is commented out until error handler is added to
 	// output scanner in the script runner. Use smaller size for now.
 	metadataMaxLength = 32768
 )
 
-var shutdownScript = fmt.Sprintf(shutdownScriptTemplate, shutdownContent, shutdownOutputPath)
-var startupScript = fmt.Sprintf(startupScriptTemplate, startupContent, startupOutputPath)
-var daemonScript = fmt.Sprintf(daemonScriptTemplate, daemonOutputPath)
+type metadataScript struct {
+	description    string
+	url            bool
+	windows        bool
+	metadataKey    string
+	scriptTemplate string
+	outputPath     string
+	outputContent  string
+	extension      string
+}
+
+// A list of all metadata script types. Windows scripts should be listed
+// in the expected run order. Startup URL scripts are excluded because
+// they are used by the test runner and implicitly tested.
+var metadataScripts = []metadataScript{
```

Comment:
you might want to put these in 2 top level structures and use a map[string]metadataScript with the key name as the map key. then you could choose to iterate over key names, as you're using this a few times below.

Reply:
Done